### DISCUSSION TITLE
[Question][Hw5] Question regarding Llvm operations

### DISCUSSION THREAD

**Discussion by PSuHyeon** [prosyslab-classroom/cs524-program-analysis#330] [Q&A]
Hi, 
I am having hard time handling Call instruction...
- Is there any Llvm API for getting function that is called in the instruction having Llvm.Opcode.Call ?
 I used to use `Llvm.operand` to get operands in the instruction, but `Llvm.operand` operation seems to only works when I use function pointer in function call. Basically in `  call void @k(i32* %call1, i32 3), !dbg !88` for example, I want to get llvalue of function K. I found out that there is `Llvmutils.function_name` operation but it returns string of function that is called. 

- Similarly how can I get arguments for the function calls? The position of operands seems to different when I use function pointer to call function with calling with function name. 

- Lastly, can the function have multiple arguments in this assignment?

Sorry for asking syntax related question. 
Thank you in advance.

**Comment by goodtaeeun** [comment]
Hi,

1. You can get the callee function by using the `Llvm.operand`. By the way, calling the function by a function pointer or by its name will bring no difference. If you look at the LLVM IR generated from `test/example5.c`, you will see that the function call by function pointer `fp` and function name `h` results in the same format of LLVM instruction.  If you have seen some cases where it is different, please provide some example codes.

2. You can get the arguments of the function call by using the `Llvm.operand`.

3. Yes, functions can have multiple arguments in this assignment.

Hope this answer helps you.
Feel free to ask further questions.

Regards,
Tae Eun.

**Comment by PSuHyeon** [comment]
I found out what was wrong in my code. Thank you for detailed explanation!
